FWCore/Utilities/interface/ConstRange.h

```diff
@@ -0,0 +1,27 @@
+#ifndef FWCore_Utilities_ConstRange_h
+#define FWCore_Utilities_ConstRange_h
+
+namespace edm
+{
+     /*
+      *class which implements begin() and end() to use range-based loop with pairs of iterators
+      */
+
+     template<class T>
+     class ConstRange
+     {
+       public:
+         using Iterator = typename T::const_iterator;
```

Comment:
Yes that would give more freedom, but then can you please explain me why you intended the name `ConstRange<T>`? I thought it was because it was supposed to use `const_iterator`. Thanks!

Reply:
Because `begin()` and `end()` would be const (even if the underlying iterator itself would be non-const).

Reply:
Ok! I have change that now.

Reply:
I think the `Const` in the name is misleading now that it takes the iterator type, not the container.

As for placement, `FWCore/Utilities` would probably be best for now.

Reply:
I agree (now) that the class should probably be called just `Range`.